server/routes/offenceCodeDecisions/offenceCodeDecisions.ts

```diff
@@ -0,0 +1,84 @@
+import { Request, Response } from 'express'
+import url from 'url'
+import decisionTrees from '../../offenceCodeDecisions/DecisionTrees'
+import { FormError } from '../../@types/template'
+import validateForm from './offenceCodeDecisionsValidation'
+import PlaceOnReportService from '../../services/placeOnReportService'
+
+import type IncidentRole from '../../offenceCodeDecisions/IncidentRoles'
+import { properCaseName } from '../../utils/utils'
+
+type PageData = {
+  error?: FormError
+  selectedDecisionId?: string
+}
+
+export default class OffenceCodeRoutes {
+  constructor(private readonly placeOnReportService: PlaceOnReportService) {}
+
+  private async placeholderValues(adjudicationNumber: string, res: Response) {
+    const { user } = res.locals
+    const draftAdjudication = await this.placeOnReportService.getDraftAdjudicationDetails(
+      Number(adjudicationNumber),
+      user
+    )
+    const prisonerDetails = await this.placeOnReportService.getPrisonerDetails(
+      draftAdjudication.draftAdjudication.prisonerNumber,
+      user
+    )
+    return {
+      offenderFirstName: properCaseName(prisonerDetails.firstName),
+      offenderLastName: properCaseName(prisonerDetails.lastName),
+    }
+  }
+
+  private renderView = async (req: Request, res: Response, pageData?: PageData): Promise<void> => {
+    const { adjudicationNumber, incidentRole } = req.params
+    const { error, selectedDecisionId } = pageData
+    const placeholderValues = await this.placeholderValues(adjudicationNumber, res)
+    const path = req.path.replace(`/${adjudicationNumber}/${incidentRole}/`, '')
+    const decision = decisionTrees.get(incidentRole as IncidentRole).findByUrl(path)
+
+    const pageTitle = decision.getTitle().getProcessedText(placeholderValues)
+    const questions = decision.getChildren().map(d => {
+      return {
+        id: d.id(),
+        label: d.getQuestion().getProcessedText(placeholderValues),
+      }
+    })
+    return res.render(`pages/${decision.getPage() || 'offenceCodeDecisions'}`, {
+      errors: error ? [error] : [],
+      selectedDecisionId,
+      questions,
+      pageTitle,
+      pageData,
+    })
+  }
+
+  view = async (req: Request, res: Response): Promise<void> => this.renderView(req, res, {})
+
+  submit = async (req: Request, res: Response): Promise<void> => {
+    const { adjudicationNumber, incidentRole } = req.params
+    const { selectedDecisionId } = req.body
+
+    const error = validateForm({ selectedDecisionId })
```

Comment:
Still plenty of stuff to do regarding getting additional info e.g. questions where we search for other prisoners

Reply:
I figure that we'll stick that in the session - its possible we could do it with url parameters but I'm thinking that way will be tricky....

Reply:
Yeah session.. the only other options are POSTing and cookies, but let's discuss that later